temporal-sdk/src/main/java/io/temporal/internal/client/LazyWorkflowUpdateHandleImpl.java

```diff
@@ -120,6 +113,34 @@ public CompletableFuture<T> getResultAsync(long timeout, TimeUnit unit) {
             });
   }
 
+  @Override
+  public T getResult() {
+    try {
+      return getResultAsync().get();
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    } catch (ExecutionException e) {
+      Throwable cause = e.getCause();
+      throw (cause instanceof RuntimeException
+          ? (RuntimeException) cause
+          : new RuntimeException(cause));
+    }
+  }
+
+  @Override
+  public T getResult(long timeout, TimeUnit unit) {
```

Comment:
No because of https://github.com/temporalio/features/issues/483 we return a `WorkflowUpdateTimeoutOrCancelledException`, you can see this in the test `WorkflowUpdateGetResultTimeout`

Reply:
Makes sense (may be worth documenting, but don't have to right now)